app/uk/gov/nationalarchives/omega/editorial/models/RelatedMaterial.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022 The National Archives
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.gov.nationalarchives.omega.editorial.models
+
+import play.api.libs.json._
+import play.api.libs.functional.syntax._
+import play.api.libs.json.Reads._
+
+sealed abstract class RelatedMaterial
+
+object RelatedMaterial {
```

Comment:
I like how you've done this. 

I would have expected a case class with two optional fields:  an object for the link with two strings, and a string for the description. 

But maybe this is better?

Reply:
I thought about a single case class with two optional fields - one field for the link text and href and another for the description.  My issue would be the case where both of those fields were `None` - to me that is allowing an unintended state to be represented.

I think that parsing json would be just as complicated in that case as well. The only case where the default json parser could be used is if all three fields were present and optional on a single case class, but that would allow several unintended states to be represented.

Reply:
That's a good point: that we're preventing an invalid state by modelling it this way.